README.md

```diff
@@ -13,5 +13,14 @@ In common:
 
 Differences:
 
-1. `program`: A ZkProgram is used to represent the proof of the ECDSA signature verification, and the proof of that program is passed to the zkApp contracts as a recursive proof. This approach is simplest but the ECDSA circuit is recursively verified every time.
-2. `merkle`: A zkApp uses an o1js `MerkleMap` (Merkle tree of depth 256) to remember the set of all delegation orders that have been verified. Inserts to this map check the ECDSA signature of the submitted order. The 'user' zkApp then accepts a `MerkleWitness` instead of a recursive proof and makes a contract-to-contract call to query whether that witness describes a delegation order whose signature was previously verified. This approach requires the delegation contract's full history to be pulled from an archive node before being able to make inserts or queries.
+1. `program`: A ZkProgram is used to represent the proof of the ECDSA signature verification, and the proof of that program is passed to the zkApp contracts as a recursive proof. This approach is simplest but the ECDSA circuit is recursively verified every time (the recursive proof can be cached).
+2. `merkle`: A zkApp uses an o1js `MerkleMap` (Merkle tree of depth 256) to remember the set of all delegation orders that have been verified. Inserts to this map check the ECDSA signature of the submitted order. The 'user' zkApp then accepts a `MerkleWitness` instead of a recursive proof and makes a contract-to-contract call to query whether that witness describes a delegation order whose signature was previously verified. This approach requires the delegation contract's full history to be pulled from an archive node before being able to make inserts or queries, but allows the delegation state to be easily queried from other contracts without having to embed the ZkProgram inside their code.
```

Comment:
> but allows the delegation state to be easily queried from other contracts without having to embed the ZkProgram inside their code.

I think this isn't accurate. In either case the client does need some pieces of the library code.

* `program`: Needs ZkProgram JS to make proofs, needs `compile()` output to verify them (easiest way to get this is to start with the JS).
* `merkle`: Needs MerkleTree SmartContract JS to make proofs that call into it.